### PR TITLE
245punc

### PR DESCRIPTION
**Pull request by nvolk**
Minor update: Fixes for field 24X punctuation rules (as discussed in Slack in 2023-10-31) and 7XX sorting. The latter prefers $t-less 7XX fields. Also, 711 used $j as relator term, instead of $e.

### CODE REVIEW

File: src/punctuation2.js [hunk 1]
```diff
@@ -394,18 +419,12 @@ function checkRule(rule, field, subfield1, subfield2) {
     return false;
   }
 
-  //nvdebug(`${name}: ACCEPT ${rule.code}/${subfield1.code}, SF2=${rule.followedBy}/${subfield2 ? subfield2.code : '#'}`, debug);
+  nvdebug(`${rule.name ? rule.name : '<unnamed>'}: ACCEPT ${rule.code} (${subfield1.code}), SF2=${rule.followedBy} (${subfield2 ? subfield2.code : '#'})`, debug);
```
Comment (natlibfi-jonollil) [inline src/punctuation2.js:422]: comment?

File: src/punctuation2.js [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
 */
 import {validateSingleField} from './ending-punctuation';
-// import createDebugLogger from 'debug';
-import {fieldToString, nvdebug} from './utils';
+import createDebugLogger from 'debug';
+import {fieldToString, nvdebug, subfieldToString} from './utils';
 import clone from 'clone';
 
-// const debug = createDebugLogger('@natlibfi/marc-record-validators-melinda/punctuation2');
+const debug = createDebugLogger('@natlibfi/marc-record-validators-melinda/punctuation2');
```
Comment (natlibfi-jonollil) [inline src/punctuation2.js:17]: Could this be changed to something like 
`const debug = createDebugLogger('debug/punctuation2');` so it wont be triggered by `@natlibfi/*` where flow debug is?

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: src/punctuation2.js (hunk 1 above)
Comment (nvolk) [reply src/punctuation2.js:422]: Punctuation rules can have 'name', which might make things easier to remember/debug, but 'name' is optional.

File: src/punctuation2.js (hunk 2 above)
Comment (nvolk) [reply src/punctuation2.js:17]: Yes.

Comment (nvolk) [reply src/punctuation2.js:17]: (I've only seen createDebugLogger used in @natlibfi style, so I have followed that convention here.

File: src/punctuation2.js (hunk 1 above)
Comment (natlibfi-jonollil) [reply src/punctuation2.js:422]: Srry about bad articulation. Should this be commented out?

Comment (nvolk) [reply src/punctuation2.js]: Commented out

File: src/punctuation2.js (hunk 2 above)
Comment (nvolk) [reply src/punctuation2.js]: Commented out two debug messages, and this became irrelevant (for now). Also changed the out-commented createdDebugLogger() function call,

**Review by ammsalme — APPROVED** [review]

**Pull request merged**